dockerfile

```
@@ -1,25 +1,31 @@
+COPY src/go.mod src/go.sum ./
+RUN go mod download
+
+# Copy the remaining source code
+COPY src/ .
+
+# Build the Go app with CGO enabled for SQLite support
+RUN CGO_ENABLED=1 GOOS=linux go build -a -ldflags '-linkmode external -extldflags "-static"' -o main ./cmd/api
     
     # ---------- Final Stage ----------
    # Start a new stage from scratch
 FROM alpine:latest
```

Hadolint (2.12.0)

[warning] Using latest is prone to errors if the image will ever update. Pin the version explicitly to a release tag

(DL3007)